pkg/controllers/monitoring-stack/controller.go

```diff
@@ -164,7 +166,41 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 		}
 	}
+	return r.cleanupResources(ctx, ms)
```

Comment:
wouldn't this run all patchers unnecessarily when a monitoring stack is deleted? 
Shouldn't we handle deletion way early on in the reconcile loop .. i.e as soon as get the `ms` - MonitoringStack ?

Reply:
If we move it to the top, when the reconcile loop runs for the first time, the stack is not completely initialised. 
Shouldn't we be adding the finalizer only after all components have been created?

Reply:
I think the idea is to run this code at the top when the deletion timestamp is set. So the algorithm would be:
* If deletion timestamp is set, delete datasource, remove finalizer and update stack
* else If finalizer not set, set finalizer and update stack
* else reconcile child resources

Reply:
Done